config_db.go

```diff
@@ -136,95 +136,169 @@ func byIDs(ids []string) bson.M {
 }
 
 // getSectionsBSON returns the config documents from the database as a slice of [bson.Raw].
-// Configuration is always fetched from the shared database. If includeOverrides is true
-// the config documents are first fetched from the local database and only fetched from
-// the shared database when they're missing from the local database. This means that a
-// local documents will override the same document from the shared database.
+// If no shared database exists all configuration is fetched from the local database. If a
+// shared database exists, configuration is fetched from the shared database, save for the
+// overrides config document which is fetched from the local database and, if [includeOverrides],
+// the overrides in the overrides config document are applied to the rest of the configuration.
 func getSectionsBSON(ctx context.Context, ids []string, includeOverrides bool) ([]bson.Raw, error) {
-	missingIDs := ids
+	db := GetEnvironment().DB()
+	if sharedDB := GetEnvironment().SharedDB(); sharedDB != nil {
```

Comment:
Not a comment about the code, but could you add comments to the `SharedDB` and `DB` functions exposed by the Environment interface and at the implementation in `envState`? I had to read through some usages to figure out what they mean and the difference. Also, for the new godocs you wrote you mention "local database" vs "shared database", is `DB` and `SharedDB` those respectively? If so could you include that in the comment of those functions?

Reply:
done